remote_list.go

```diff
@@ -119,14 +119,17 @@ func NewHostnameResults(ctx context.Context, l *logrus.Logger, d time.Duration,
 		ticker := time.NewTicker(d)
 		go func() {
 			defer ticker.Stop()
+			var lookupSuccess bool
 			for {
+				lookupSuccess = true
 				netipAddrs := map[netip.AddrPort]struct{}{}
 				for _, hostPort := range r.hostnames {
 					timeoutCtx, timeoutCancel := context.WithTimeout(ctx, r.lookupTimeout)
 					addrs, err := net.DefaultResolver.LookupNetIP(timeoutCtx, r.network, hostPort.name)
 					timeoutCancel()
 					if err != nil {
 						l.WithFields(logrus.Fields{"hostname": hostPort.name, "network": r.network}).WithError(err).Error("DNS resolution failed for static_map host")
+						lookupSuccess = false
```

Comment:
I think for correctness, the default would need to be false and toggle this to `true` when err == nil. This is because if we try to resolve multiple hostnames for a given Nebula IP, and one fails and another succeeds, we'd like to use the results from the successful query.